core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -33,4 +43,48 @@ static void checkMetalakeExists(EntityStore store, String metalake)
       throw new RuntimeException(e);
     }
   }
+
+  static RoleEntity getRoleEntity(NameIdentifier identifier) {
+
+    Cache<NameIdentifier, RoleEntity> cache =
+        GravitinoEnv.getInstance().accessControlManager().getRoleCache();
+    EntityStore store = GravitinoEnv.getInstance().entityStore();
+
+    return cache.get(
+        identifier,
+        id -> {
+          try {
+            return store.get(identifier, Entity.EntityType.ROLE, RoleEntity.class);
+          } catch (IOException ioe) {
+            LOG.error("getting roles {} failed  due to storage issues", identifier, ioe);
+            throw new RuntimeException(ioe);
+          }
+        });
+  }
+
+  static List<RoleEntity> getValidRoles(
+      String metalake, List<String> roleNames, List<Long> roleIds) {
+    List<RoleEntity> roleEntities = Lists.newArrayList();
+    if (roleNames == null || roleNames.isEmpty()) {
+      return roleEntities;
+    }
+
+    int index = 0;
+    for (String role : roleNames) {
+      try {
+
+        RoleEntity roleEntity =
+            AuthorizationUtils.getRoleEntity(NameIdentifierUtils.ofRole(metalake, role));
+
+        if (roleEntity.id().equals(roleIds.get(index))) {
+          roleEntities.add(roleEntity);
+        }
+        index++;
+
+      } catch (NoSuchEntityException nse) {
+        // ignore this entity
+      }
+    }
+    return roleEntities;
+  }
```

Comment:
Also why don't you put this in `RoleManager`?

Reply:
Done.